adapters/hono-adapter-amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -0,0 +1,284 @@
+
+        config = new ProtocolAdapterProperties();
+        config.setAuthenticationRequired(false);
+        config.setInsecurePort(4040);
+    }
+
+    /**
+     * Verifies that a client provided Proton server instance is used and started by the adapter instead of
+     * creating/starting a new one.
+     * 
+     * @param ctx The test context to use for running asynchronous tests.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testStartUsesClientProvidedAmqpServer(final TestContext ctx) {
+        // GIVEN an adapter with a client provided Amqp Server
+        final ProtonServer server = getAmqpServer();
+        final VertxBasedAmqpProtocolAdapter adapter = getAdapter(server);
+
+        // WHEN starting the adapter
+        final Async startup = ctx.async();
+        Future<Void> startupTracker = Future.future();
+        startupTracker.setHandler(ctx.asyncAssertSuccess(result -> {
+            startup.complete();
+        }));
+        adapter.start(startupTracker);
+
+        // THEN the client provided server is started
+        startup.await();
+        verify(server).connectHandler(any(Handler.class));
+        verify(server).listen(any(Handler.class));
+    }
+
+    /**
+     * Verifies that the AMQP Adapter does not support anonymous relay.
+     */
+    @Test
+    public void anonymousRelyNotSupported() {
+        // GIVEN an AMQP adapter with a configured server.
+        final ProtonServer server = getAmqpServer();
+        final VertxBasedAmqpProtocolAdapter adapter = getAdapter(server);
+
+        // WHEN the adapter receives a request from a link
+        // that does not specify a target address
+        ProtonReceiver receiver = mock(ProtonReceiver.class);
+        adapter.handleRemoteReceiverOpen(receiver, mock(ProtonConnection.class));
+
+        // THEN the adapter rejects the message and closes the link.
+        verify(receiver).close();
+    }
+
+    /**
+     * Verifies that the adapter fails the upload of a telemetry message if the tenant for which a device belongs to has
+     * been disabled.
+     */
+    @Test
+    public void testUploadTelemetryFailsForDisabledTenant() {
+        // Given an adapter
+        final ProtonServer server = getAmqpServer();
+        MessageSender sender = givenAmessageSenderForAnyTenant();
+
+        // THAT is disabled for a tenant
+        givenAConfiguredTenant("disabled-tenant", false);
+        final VertxBasedAmqpProtocolAdapter adapter = getAdapter(server);
+
+        // When a device that belongs to such a tenant uploads a telemetry message
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+
+        final Message message = getFakeMessage();
+        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", "disabled-tenant", "the-device");
+        AmqpContext context = spy(new AmqpContext(delivery, message, resource));
```

Comment:
final

Reply:
Should we add `final` to checkstyle?

Reply:
> Should we add final to checkstyle?

I believe so.

> maybe also adding license/copyright headers to checkstyle too?

One step after the other ;-) But seriously, I am not sure if we can come up with a good regex that is both lenient enough to allow for slight variations but strict enough to find the real positives.

Reply:
> > Should we add final to checkstyle?
> I believe so.

Ok I will work on this.